zerver/views/streams.py

```diff
@@ -617,9 +617,16 @@ def add_subscriptions_backend(
             group_settings_map[setting_name] = get_stream_permission_default_group(
                 setting_name, system_groups_name_dict, creator=user_profile
             )
-            setting_groups_dict[group_settings_map[setting_name].id] = group_settings_map[
-                setting_name
-            ].id
+            if permission_configuration.default_group_name == "stream_creator_or_nobody":
```

Comment:
We could probably also make this code more generic by checking if setting group is anonymous group and get the members and subgroups of the anonymous group, but that would mean extra DB queries.

This can be fine for now and will handle all future settings with `stream_creator_or_nobody` default.